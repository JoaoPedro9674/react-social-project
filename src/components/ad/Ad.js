import React, { PropTypes } from 'react';
import { AutoAffix } from 'react-overlays';

const Ad = (props) => {
  return (
    <AutoAffix viewportOffsetTop={props.offset} container={this}>
      <div className="ad">
        <a target="_blank" rel="noreferrer noopener" href={props.url}>
          <img className="img-responsive" src={props.imageUrl} alt="React in Action by Mark Thomas | Manning Publications" />
        </a>
        <small>ads by Letters</small>
      </div>
    </AutoAffix>
  );
};

Ad.propTypes = {
  imageUrl: PropTypes.string,
  url: PropTypes.string,
  offset: PropTypes.number,
};

export { Ad };
