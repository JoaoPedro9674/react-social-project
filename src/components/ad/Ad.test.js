import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { Ad } from './Ad';

describe('<Ad/>', () => {
  it('should render correctly', () => {
    const component = renderer.create(<Ad />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('should render an image', () => {
    const wrapper = shallow(<Ad />);
  });
});
