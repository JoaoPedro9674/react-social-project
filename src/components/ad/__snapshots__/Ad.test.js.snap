exports[`<Ad/> should render correctly 1`] = `
<div
  className="ad">
  <a
    href={undefined}
    rel="noreferrer noopener"
    target="_blank">
    <img
      alt="React in Action by Mark Thomas | Manning Publications"
      className="img-responsive"
      src={undefined} />
  </a>
  <small>
    ads by Letters
  </small>
</div>
`;

exports[`<Ad/> should show a login option with user state 1`] = `
<div
  className="ad">
  <a
    href={undefined}
    rel="noreferrer noopener"
    target="_blank">
    <img
      alt="React in Action by Mark Thomas | Manning Publications"
      className="img-responsive"
      src={undefined} />
  </a>
  <small>
    ads by Letters
  </small>
</div>
`;
